### ISSUE TITLE
Question on the normalization dimension of ONNX Softmax op 

### ISSUE THREAD

**Issue by Jonxjdong** [onnx/onnx#4605]
# Ask a Question

### Question
According to the operator document, the normalization of `Softmax `should be done on the nominated axis dimension: 

### https://github.com/onnx/onnx/blob/main/docs/Operators.md#Softmax

> _The "axis" attribute indicates the dimension along which Softmax will be performed. The output tensor has the same shape and contains the Softmax values of the corresponding input._

While when I was using OnnxRuntime to do inference on the model "ResNet101-DUC-12.onnx",  which has the `output `of dimension (1, 19, 160000) from the last operation "Softmax" with axis=1, I got the output which seems normalized from the whole **axis (0, 1, 2)**, rather than the nominated **axis = 1** .
I mean that the output I got with value `np.sum(output) = 1`, however from the definition, it should be `np.sum(output[0, :, 0]) = 1`. 

Why do sometimes the behavior of `Softmax `is diffrent from the definition with N-dimension input(N>2) ?

The model above is from model zoo: 
https://github.com/onnx/models/blob/main/vision/object_detection_segmentation/duc/model/ResNet101-DUC-12.onnx

**Comment by jcwchen** [comment]
Hi @Jonxjdong,
The description you are looking at is for the latest Softmax-13. Take [ResNet101-DUC-12.onnx](https://github.com/onnx/models/blob/main/vision/object_detection_segmentation/duc/model/ResNet101-DUC-12.onnx) as an example, the opset_version is 12 so its Softmax should be Softmax-11 ([document](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#softmax-11)) instead. IIRC, there are some behavior changes between Softmax-11 and Softmax-13. Does Softmax-11 meet what you have observed?

**Comment by Jonxjdong** [comment]
Thank you. I got the difference. 
It seems Softmax behavior with the version before 13 does not keep align with tensorflow and pytorch, while the new version fix it. The DUC model from model zoo follow the old behavior.

**Comment by jcwchen** [comment]
> It seems Softmax behavior with the version before 13 does not keep align with tensorflow and pytorch, while the new version fix it. The DUC model from model zoo follow the old behavior.

Yes, that's why https://github.com/onnx/onnx/pull/2879 ONNX updated old Softmax to Softmax-13 in order to make it align with PyTorch and TensorFlow.
